client/blocks/upload-image/index.jsx

```diff
@@ -226,14 +229,18 @@ class UploadImage extends Component {
 
 		const classes = classnames( 'upload-image-modal', additionalImageEditorClasses );
 
+		const isEditingDefaultImage = defaultImage && selectedImage === defaultImage.URL;
+
+		const media = isEditingDefaultImage ? defaultImage : {
+			src: selectedImage,
```

Comment:
Does the `defaultImage` really have a `src` attribute? It's an object with this shape:
<img width="719" alt="media-item-shape" src="https://user-images.githubusercontent.com/664258/28958321-301efd4c-78f6-11e7-8bfe-b35642e6bf5c.png">

`media` has a different shape/type and some conversion needs to happen, doesn't it?

When reading the code, I was a bit confused about what type of value the `defaultImage` is expected to be. Some documentation, or comments, or specifying something better that `defaultImage: PropTypes.any` in `propTypes` (`PropTypes.shape` and `PropTypes.oneOf` give us nice tools to do that) would make it much clearer.